include/sparrow/array/data_traits.hpp

```diff
@@ -39,83 +41,23 @@ namespace sparrow
         using default_layout = null_layout<DS>;
     };
 
-    template <>
-    struct arrow_traits<bool> : common_native_types_traits<bool>
-    {
-        static constexpr data_type type_id = data_type::BOOL;
-    };
-
-    template <>
-    struct arrow_traits<std::uint8_t> : common_native_types_traits<std::uint8_t>
-    {
-        static constexpr data_type type_id = data_type::UINT8;
-    };
-
-    template <>
-    struct arrow_traits<std::int8_t> : common_native_types_traits<std::int8_t>
-    {
-        static constexpr data_type type_id = data_type::INT8;
-    };
-
-    template <>
-    struct arrow_traits<char> : common_native_types_traits<char>
-    {
-        static constexpr data_type type_id = data_type::UINT8;
-    };
-
-    template <>
-    struct arrow_traits<std::uint16_t> : common_native_types_traits<std::uint16_t>
-    {
-        static constexpr data_type type_id = data_type::UINT16;
-    };
-
-    template <>
-    struct arrow_traits<std::int16_t> : common_native_types_traits<std::int16_t>
-    {
-        static constexpr data_type type_id = data_type::INT16;
-    };
-
-    template <>
-    struct arrow_traits<std::uint32_t> : common_native_types_traits<std::uint32_t>
-    {
-        static constexpr data_type type_id = data_type::UINT32;
-    };
-
-    template <>
-    struct arrow_traits<std::int32_t> : common_native_types_traits<std::int32_t>
-    {
-        static constexpr data_type type_id = data_type::INT32;
-    };
-
-    template <>
-    struct arrow_traits<std::uint64_t> : common_native_types_traits<std::uint64_t>
-    {
-        static constexpr data_type type_id = data_type::UINT64;
-    };
-
-    template <>
-    struct arrow_traits<std::int64_t> : common_native_types_traits<std::int64_t>
+    // Define automatically all standard integral types support (including `bool`).
+    template <class T>
+        requires std::integral<T>
+    struct arrow_traits<T> : common_native_types_traits<T>
     {
-        static constexpr data_type type_id = data_type::INT64;
+        static constexpr data_type type_id = data_type_from_size<T>();
     };
 
-    template <>
-    struct arrow_traits<float16_t> : common_native_types_traits<float16_t>
-    {
-        static constexpr data_type type_id = data_type::HALF_FLOAT;
-    };
 
-    template <>
-    struct arrow_traits<float32_t> : common_native_types_traits<float32_t>
+    // Define automatically all standard floating-point types support.
+    template<class T>
+        requires std::floating_point<T>
+    struct arrow_traits<T> : common_native_types_traits<T>
     {
-        static constexpr data_type type_id = data_type::FLOAT;
+        static constexpr data_type type_id = data_type_from_size<T>();
     };
```

Comment:
Can't you merge with the previous definition by merging the requires or creating an "arithmetic" concept ?

Reply:
You mean using `std::integer<T> or std::floating_point<T>` ? Yes that would make sense, although I'll refrene from making an arithmetic concept for now because I suspect it implies additional expectations (not sure) but I can at least merge both specialisations with these constraintes yes, will do.